as/thrift.js

```diff
@@ -41,15 +41,18 @@ function TChannelAsThrift(opts) {
 
     var self = this;
 
-    assert(opts && typeof opts.source === 'string',
-        'must pass source as an argument');
+    assert(opts, 'options required');
+    assert(typeof opts === 'object', 'options must be object');
+    assert(
+        typeof opts.source === 'string' ||
+        typeof opts.entryPoint === 'string',
+        'must pass source as an argument'
+    );
 
-    self.thriftSource = opts.source;
-    self.thriftFileName = opts.thriftFileName || 'service.thrift';
     self.spec = new thriftrw.Thrift({
-        entryPoint: opts.entryPoint,
+        entryPoint: opts.entryPoint || opts.thriftFileName || 'service.thrift',
```

Comment:
Is the 'service.thrift' default necessary? In the absence of an entryPoint, ThriftRW defaults the entryPoint property.